### PR TITLE
Fix types warnings

### LINKED ISSUE #121: Export all forgotten types
This entire task involves adding exports to `src/index.ts` for the following values.

After completed, the script `npm run build:readme` should run with zero warnings. Here are the current warnings:

```sh
Warning: Modify, defined at src/types/Block.types.ts:3, is referenced by BlockResponse but not included in the documentation.
Warning: RPCBlock, defined at src/types/Block.types.ts:20, is referenced by BlockResponse but not included in the documentation.
Warning: Modify, defined at src/types/Transaction.types.ts:3, is referenced by TransactionResponse but not included in the documentation.
Warning: RPCTransaction, defined at src/types/Transaction.types.ts:4, is referenced by TransactionResponse but not included in the documentation.
Warning: TransactionReceipt, defined at src/types/Transaction.types.ts:34, is referenced by FallthroughProvider.getTransactionReceipt.getTransactionReceipt but not included in the documentation.
Warning: BlockTransactionResponse, defined at src/types/Transaction.types.ts:64, is referenced by BlockResponse.__type.transactions but not included in the documentation.
Warning: ConstructorOptions, defined at src/providers/FallthroughProvider.ts:13, is referenced by FallthroughProvider.constructor.new FallthroughProvider.options but not included in the documentation.
Warning: BlockTag, defined at src/types/Block.types.ts:46, is referenced by FallthroughProvider.getBalance.getBalance.blockTag but not included in the documentation.
```

### PR DESCRIPTION
**Pull request by icepy**
Closes #121 

typedoc default only exports from the direct entrypoints are included

- [typedoc #1739](https://github.com/TypeStrong/typedoc/issues/1739)
- [typedoc #1657](https://github.com/TypeStrong/typedoc/issues/1657)

i used this plugin.

[https://github.com/Gerrit0/typedoc-plugin-missing-exports](https://github.com/Gerrit0/typedoc-plugin-missing-exports)

### CODE REVIEW

**Review by arimgibson — APPROVED** [review]
This looks great, thanks @icepy ! @dawsbot is ultimately in charge of approving and merging 😁

File: tsconfig.json [hunk 1]
```diff
@@ -19,5 +19,10 @@
   },
   "include": ["./src"],
   "types": ["node", "jest"],
-  "exclude": ["src/**/*.test.ts", "markdown.config.js"]
+  "exclude": ["src/**/*.test.ts", "markdown.config.js"],
+  "typedocOptions": {
+    "entryPoints": ["src/index.ts"],
+    "json": "scripts/markdown-magic/typedoc.out.json",
+    "plugin": ["typedoc-plugin-missing-exports"]
+  }
```
Comment (arimgibson) [inline tsconfig.json:27]: `pretty` option is enabled by default, could be expressly defined though

Comment (icepy) [reply tsconfig.json:27]: that's great!

**Comment by Gerrit0** [comment]
While this does remove the warnings, it does not really address the reason that the warnings were there in the first place. If a user of this package wants to reference `ConstructorOptions` to construct an object and pass it to a few of their functions before creating a `FallthroughProvider`, they have a few choices:

1. Import the type from the internal path, and hope it never changes: `import { ConstructorOptions } from "essential-eth/lib/esm/providers/FallthroughProvider"`
2. Derive the type from the exported type: `type ConstructorOptions = ConstructorParameters<typeof FallthroughProvider>[1]`
3. Copy the definition and manually keep it up to date when updating the library.

File: src/providers/FallthroughProvider.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ const promiseTimeout = (prom: Promise<any>, time: number) =>
     ),
   ]);
 
-interface ConstructorOptions {
+export interface ConstructorOptions {
```
Comment (arimgibson) [inline src/providers/FallthroughProvider.ts:13]: @dawsbot I figured this type was worth exporting, but it can be ignored if not

**Comment by arimgibson** [comment]
@dawsbot take a look at my most recent commit when ya get a chance. I went through it manually; i.e. exported some of the missing types and ignored others that shouldn't be exported (used internally). 

Couple things I found along the way worth mentioning to explain why I did what I did:
- Using the `@ignore` tag before a type/function/etc that you are purposely not exporting will still throw an error if that type/function/etc is referenced by something that is being exported
- Same with `@internal` even with the `--excludeInternal` option set as true
- `intentionallyNotExported` doesn't support glob or regex, hence having to add `Transaction` and `Block` as separate lines

**Pull request merged**